src-self-hosted/astgen.zig

```diff
@@ -176,76 +176,52 @@ fn varDecl(
                 }
-            } else {
-                const result_loc: ResultLoc = if (node.getTrailer("type_node")) |type_node|
-                    .{ .ty = try typeExpr(mod, scope, type_node) }
+            } else r: {
+                if (node.getTrailer("type_node")) |type_node|
+                    break :r ResultLoc{ .ty = try typeExpr(mod, scope, type_node) }
                 else
-                    .none;
-                const init_inst = try expr(mod, scope, result_loc, init_node);
-                const sub_scope = try block_arena.create(Scope.LocalVal);
-                sub_scope.* = .{
-                    .parent = scope,
-                    .gen_zir = scope.getGenZIR(),
-                    .name = ident_name,
-                    .inst = init_inst,
-                };
-                return &sub_scope.base;
-            }
+                    break :r .none;
+            };
+            const init_inst = try expr(mod, scope, result_loc, init_node);
+            const sub_scope = try block_arena.create(Scope.LocalVal);
+            sub_scope.* = .{
+                .parent = scope,
+                .gen_zir = scope.getGenZIR(),
+                .name = ident_name,
+                .inst = init_inst,
+            };
+            return &sub_scope.base;
         },
         .Keyword_var => {
-            if (node.getTrailer("type_node")) |type_node| {
+            const alloc = if (node.getTrailer("type_node")) |type_node| a: {
                 const type_inst = try typeExpr(mod, scope, type_node);
-                const alloc = try addZIRUnOp(mod, scope, name_src, .alloc, type_inst);
-                const result_loc: ResultLoc = .{ .ptr = alloc };
-                const init_inst = try expr(mod, scope, result_loc, init_node);
-                const sub_scope = try block_arena.create(Scope.LocalPtr);
-                sub_scope.* = .{
-                    .parent = scope,
-                    .gen_zir = scope.getGenZIR(),
-                    .name = ident_name,
-                    .ptr = alloc,
-                };
-                return &sub_scope.base;
-            } else {
-                const alloc = try addZIRNoOp(mod, scope, name_src, .alloc_inferred);
-                const result_loc = .{ .inferred_ptr = alloc.castTag(.alloc_inferred).? };
-                const init_inst = try expr(mod, scope, result_loc, init_node);
-                const sub_scope = try block_arena.create(Scope.LocalPtr);
-                sub_scope.* = .{
-                    .parent = scope,
-                    .gen_zir = scope.getGenZIR(),
-                    .name = ident_name,
-                    .ptr = alloc,
-                };
-                return &sub_scope.base;
-            }
+                break :a try addZIRUnOp(mod, scope, name_src, .alloc, type_inst);
+            } else try addZIRNoOp(mod, scope, name_src, .alloc_inferred);
+            const result_loc = r: {
```

Comment:
this introduces 2 branches on the type node rather than status quo of 1. Personally I find status quo significantly easier to read & maintain. However I would ok with this change if you follow this pattern to get 2 values out of the branch: https://github.com/ziglang/zig/blob/4adc052f0b8fbb5c3f5ac06cc92d2fc9bd7e409e/src-self-hosted/codegen.zig#L415-L429